powerlibs/aws/sqs/dequeue_to_api/__init__.py

```diff
@@ -13,6 +13,9 @@
 
 
 class DequeueToAPI(SQSDequeuer):
+
+    request_methods = {}
```

Comment:
So each instance is going to set the class `request_methods` each time they are instantiated?
It does not smell good to me...
Besides, what's the point of this change? It doesn't seem to add anything.

Reply:
Each instance was already setting the `request_methods` at each instantiation. This doesn't change any behavior, as far as I know. But it makes the class always aware of this attr, even before instantiation, making auto-completes and mocks possible...
Actually, for this reasons I always prefer to have all classes attrs declared at top level

Reply:
Actually, having **all** attributes declared at "top level" is wrong, because these attributes are **class attributes**. It's an absolutely different concept than instance attributes.

Class attributes **belong to the class**. You shoudn't ever use a class attribute if your problem requires an instance attribute. They are different concepts used for different cases, although sometimes the effects of doing the wrong thing may be unnoticeable.

It makes no sense to have an instance method setting a class attribute like this. It's weird. It's wrong. The class attributes should be instantiated immediately or, at least, inside a `@classmethod`.

**But**, this question shed light into some issues: for example: why the hell the method `load_request_methods` receives a `requests_module` as argument? It would force anyone trying to use another "requests method" **to reimplement the entire `\__init__' method all over again**. And that's a horrible idea! This part of the code was "tainted by the tests", not based on practical requirements and uses.

So **maybe we should try a different aproach**: let's get rid of this "load request methods" idea and, instead, simply implement `get`, `post`, `put`, `patch` and `delete` methods directly into the class -- since the "ToAPI" part in the class name makes it reasonably clear that these methods are intended to perform HTTP requests. Thus the code gets much more elegant and easier to use.

Sounds right?

Reply:
Working on it